src/clj/puppetlabs/puppetserver/jruby_request.clj

```diff
@@ -45,6 +46,24 @@
      jruby-puppet jruby-service {:request (dissoc request :ssl-client-cert)}
      (handler (assoc request :jruby-instance jruby-puppet)))))
 
+(defn wrap-with-concurrent-request-limit
+  "Middleware fn that returns a 503 'Service Unavailable' response if the
+  number of outstanding requests for a JRuby instance exceed the given limit.
+  If splay-limit is set, the response will include a Retry-After header set
+  to a random fraction of the limit."
+  [handler metrics-service request-limit splay-limit]
+  ;; TODO: Log warnings when limit is hit. Add i18n to user-facing strings.
+  (fn [request]
+    (let [metrics-map (jruby-metrics/get-metrics metrics-service)
+          requested-count (-> metrics-map :requested-count .getCount)]
+      (if (> requested-count request-limit)
```

Comment:
Ahh. Good catch! `requested-instances` is definitely what I want since it looks like we don't `.dec` the counter when an instance is returned.